File: src/game/Vision.js
import { Colores } from "./Colores.js";

export const Vision = Object.freeze({
  PYRO: "PYRO",
  HYDRO: "HYDRO",
  ELECTRO: "ELECTRO",
  DENDRO: "DENDRO",
  ANEMO: "ANEMO",
  GEO: "GEO",
  CRYO: "CRYO"
});

export function visionGanaA(vision, otra) {
  switch (vision) {
    case Vision.PYRO: return otra === Vision.CRYO;
    case Vision.HYDRO: return otra === Vision.PYRO;
    case Vision.ELECTRO: return otra === Vision.HYDRO;
    case Vision.DENDRO: return otra === Vision.ELECTRO;
    case Vision.ANEMO: return otra === Vision.DENDRO;
    case Vision.GEO: return otra === Vision.ANEMO;
    case Vision.CRYO: return otra === Vision.GEO;
    default: return false;
  }
}

export function visionRandom() {
  const values = Object.values(Vision);
  return values[Math.floor(Math.random() * values.length)];
}

export function visionColor(vision) {
  switch (vision) {
    case Vision.PYRO: return Colores.PYRO;
    case Vision.HYDRO: return Colores.HYDRO;
    case Vision.CRYO: return Colores.CRYO;
    case Vision.ELECTRO: return Colores.ELECTRO;
    case Vision.DENDRO: return Colores.DENDRO;
    case Vision.ANEMO: return Colores.ANEMO;
    case Vision.GEO: return Colores.GEO;
    default: return Colores.RESET;
  }
}

File: java/src/game/Vision.java
package game;
//enum(Def W3 School): una "clase" especial, nos permite representar un grupo de constantes, normalmente finals.
public enum Vision {
    PYRO, HYDRO, ELECTRO, DENDRO, ANEMO, GEO, CRYO;
    //Condicional de criterio sobre que vision gana a la otra
    public boolean ganaA(Vision otra) {
        return switch (this) {
            case PYRO -> otra == CRYO;
            case HYDRO -> otra == PYRO;
            case ELECTRO -> otra == HYDRO;
            case DENDRO -> otra == ELECTRO;
            case ANEMO -> otra == DENDRO;
            case GEO -> otra == ANEMO;
            case CRYO -> otra == GEO;
        };
    }
    /**
     * Generar vision aleatoria para todos los personajes, esto hace que sea totalmente al azar cada resultado
     * @return una vision aleatoria por personaje
     */
    public static Vision random() {
        Vision[] v = values();
        return v[(int) (Math.random() * v.length)];
    }

    //Listado de colores para cada vision, enlazado desde la clase colores
    public String color() {
        return switch (this) {
            case PYRO -> Colores.PYRO;
            case HYDRO -> Colores.HYDRO;
            case CRYO -> Colores.CRYO;
            case ELECTRO -> Colores.ELECTRO;
            case DENDRO -> Colores.DENDRO;
            case ANEMO -> Colores.ANEMO;
            case GEO -> Colores.GEO;
        };
    }
}
